File: src/main/kotlin/com/sourcegraph/cody/agent/protocol_generated/RemoteRepoFetchState.kt
@file:Suppress("FunctionName", "ClassName", "unused", "EnumEntryName", "UnusedImport")
package com.sourcegraph.cody.agent.protocol_generated;

import com.google.gson.annotations.SerializedName;

data class RemoteRepoFetchState(
  val state: StateEnum, // Oneof: paused, fetching, errored, complete
  val error: CodyError? = null,
) {

  enum class StateEnum {
    @SerializedName("paused") Paused,
    @SerializedName("fetching") Fetching,
    @SerializedName("errored") Errored,
    @SerializedName("complete") Complete,
  }
}

File: agent/bindings/csharp/RemoteRepoFetchState.cs
using System.Text.Json.Serialization;

namespace Cody.Core.Agent.Protocol
{
  public class RemoteRepoFetchState
  {
    [JsonProperty(PropertyName = "state")]
    public StateEnum State { get; set; } // Oneof: paused, fetching, errored, complete
    [JsonProperty(PropertyName = "error")]
    public CodyError Error { get; set; }

    public enum StateEnum
    {
      [EnumMember(Value = "paused")]
      Paused,
      [EnumMember(Value = "fetching")]
      Fetching,
      [EnumMember(Value = "errored")]
      Errored,
      [EnumMember(Value = "complete")]
      Complete,
    }
  }
}
